Fix the selector field of service resource

cnbailian opened the pull request:
fix #1399 

一点小建议，因为没有详细的看过整个项目，可能不准确：我认为这个问题主要是原本在资源关系网中通用的 name 因为两种 svc，变为不通用。我建议可以将某个 name 作为关系网的不变值，例如 service.Name，而参数中的 `name string` 则可以成为每个资源自定义的 name。

cnbailian commented:
I signed it

cnbailian commented:
I signed it

hangyan commented:
@cnbailian Thanks for your contribution. Please use english later. Will review on this ASAP.

hangyan commented:
@cnbailian You may need to add some unit test to test this change.

hangyan commented:
Or you can look up the `make test-cmd` test, it's more complicated and useful.

cnbailian commented:
@hangyan OK, I'm sorry I didn't check the test cases, I'll do it

hangyan commented:
@cnbailian It's ok, just take your time. No rush

cnbailian commented:
@hangyan Hi, I ran `make test-cmd` locally and it passed the test, but there are some "FAIL" messages that I don't know how to fix.

```
===> Starting test <===
convert::expect_success: Running: 'kompose -f /Users/bailian/GoProject/kompose/script/test/fixtures/v2/docker-compose.yaml convert --stdout -j --with-kompose-annotation=false' expected_output: '/Users/bailian/GoProject/kompose/script/test/fixtures/v2/output-k8s.json'
INFO Create kubernetes pod instead of pod controller due to restart policy: no
WARN Create multiple service to avoid using mixed protocol in the same service when it's loadbalander type
PASS: converted output matches
FAIL: warnings given: INFO Create kubernetes pod instead of pod controller due to restart policy: no
WARN Create multiple service to avoid using mixed protocol in the same service when it's loadbalander type

===> Starting test <===
convert::expect_success: Running: 'kompose --provider=openshift -f /Users/bailian/GoProject/kompose/script/test/fixtures/v2/docker-compose.yaml convert --stdout -j --with-kompose-annotation=false' expected_output: '/Users/bailian/GoProject/kompose/script/test/fixtures/v2/output-os.json'
WARN Create multiple service to avoid using mixed protocol in the same service when it's loadbalander type
PASS: converted output matches
FAIL: warnings given: WARN Create multiple service to avoid using mixed protocol in the same service when it's loadbalander type
```

hangyan commented:
@cnbailian  I will look into it. You can try unittest first

hangyan commented:
/lgtm
